pkg/flow/flowtest/commands.go

```diff
@@ -0,0 +1,136 @@
+package flowtest
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/pkg/flow"
+	"github.com/grafana/dskit/backoff"
+	"rsc.io/script"
+)
+
+func createCommands(f *flow.Flow, opts *options) map[string]script.Cmd {
+	cmds := script.DefaultCmds()
+	registerControllerCommands(f, cmds)
+
+	for name, cmd := range opts.cmds {
+		cmds[name] = cmd
+	}
+
+	return cmds
+}
+
+func registerControllerCommands(f *flow.Flow, cmds map[string]script.Cmd) {
+	cmds["controller.load"] = script.Command(
+		script.CmdUsage{
+			Summary: "load a file into the Flow controller",
+			Args:    "file",
+		},
+		func(s *script.State, args ...string) (script.WaitFunc, error) {
+			if len(args) != 1 {
+				return nil, script.ErrUsage
+			}
+
+			bb, err := os.ReadFile(makeAbsolute(s.Getwd(), args[0]))
+			if err != nil {
+				return nil, err
+			}
+			src, err := flow.ParseSource(args[0], bb)
+			if err != nil {
+				return nil, fmt.Errorf("parsing %q: %w", args[0], err)
+			}
+			return nil, f.LoadSource(src, nil)
+		},
+	)
+
+	cmds["controller.start"] = script.Command(
```

Comment:
I thought it would make most sense to be as close to user experience as possible with those tests so things like quirks of CLI parsing or conflicting flag combinations or environment variables can be tested. Also, we can completely refactor things inside and validate that the users' interface remains the same.

The users call `grafana-agent run <config file>` and they don't start a controller themselves, so I would think that instead of operating on the controller level we should operate on agent's CLI level? We'd still need a way to extract controller from a running agent for the purposes of asserting on conditions like 'healthy', which is fine.